### PR TITLE
don't enable TEC with default / invalid TEC calibration

### PR DESCRIPTION
**Pull request by mzieg**

### CODE REVIEW

**Comment by samiebee43** [comment]
Looks good, except in `wasatch > utils.py > coeffs_look_valid` some of the validation steps don't make sense to me.

(1) checking for [0, 1, 0,...]

This corresponds to `y=x`, and while that indicates a situation where no calibration is applied, is it strictly invalid?

(2) checking for constants (all coeffs equal)

I think the intention is to capture `y=a` (a horizontal line) ? 

Instead the current implementation considers a variety of polynomials `y=a+ax+axx+axxx...` to be invalid. Most of these we probably never encounter.

A horizontal line calibration could be used to specify a global default for a unreliable sensor, so we may not want to consider that invalid either.

**Comment by mzieg** [comment]
> (1) checking for [0, 1, 0,...]
> This corresponds to `y=x`, and while that indicates a situation where no calibration is applied, is it strictly invalid?

I think so, yeah.  An uncalibrated TEC shouldn't be enabled, as it could overrun and damage the device.

> (2) checking for constants (all coeffs equal)
> I think the intention is to capture `y=a` (a horizontal line) ?

This would also catch a variety of cases a technician or user might use to indicate "this has not been calibrated," such as "all zeros", "all -1", etc.  No legitimate calibration is likely to have matching values.  I think this is a reasonable test.

This also automatically handles the case where the EEPROM page is unprogrammed (all bytes are 0xff) or where the EEPROM has been zeroed (all bytes are 0x00).

> Instead the current implementation considers a variety of polynomials `y=a+ax+axx+axxx...` to be invalid. Most of these we probably never encounter.

I've encountered many of them.
 
> A horizontal line calibration could be used to specify a global default for a unreliable sensor, so we may not want to consider that invalid either.

I would not recommend marking detectors unreliable by putting magic numbers into the TEC calibration.  I'm not sure we need an EEPROM field to indicate "unreliable unit," but if we do we can allocate a bit in FeatureMask or something.

Bottom line, I agree that there are various reasons why a series of floating-point polynomial calibration coefficients might be found to contain straight lines and so forth.  However, none of them will be *valid* calibrations.  They might be used to indicate all sorts of things, but the fact that they're being used as indicators and flags underscores the reality that, whatever they are, they are NOT valid calibrations.  Our spectrometers don't have any calibrations in which a straight line is a likely condition, and I think it's better to flag such cases than to automatically act on them as though they were a "real" calibration.
